### PR TITLE
Added warning message when AirloopHVAC:UnitarySystem all flow rates are zero and the unit has a fan

### LINKED ISSUE #7166: AirloopHVAC:UnitarySystem should warn if all flow rates are zero and the unit has a fan in it
Issue overview
--------------
User file had blanks for all of the `AirloopHVAC:UnitarySystem` cooling/heating/noload air flow rate fields. This resulted in no system operation, zero HVAC energy consumption, and no warnings in the error file. Seems if a unitary system has fan in it and all of the airflow rates are zero, this should prompt a warning. Likewise, if there's a cooling coil, should the cooling airflow rate be > 0? And for heating? Other combinations that can reliably throw a warning?

### Details
Some additional details for this issue (if relevant):

 - Version of EnergyPlus 9.0.1
 - Unmethours [37081](https://unmethours.com/question/37081/model-using-zero-heating-and-cooling/)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file attached here
[scenario_1.idf.txt](https://github.com/NREL/EnergyPlus/files/2843094/scenario_1.idf.txt)

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #7166
 - Added warning message when AirloopHVAC:UnitarySystem all flow sizing method and flowing sizing input fields are blank and the unit has a fan in it. Warning is issued by checking all relevant fields for cooling and heating supply air flow rates.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally

### CODE REVIEW

**Comment by Nigusse** [comment]
Sample warning message:

   ** Warning ** AirLoopHVAC:UnitarySystem = FAN COOLING COIL HEATING COIL UNITARYSYSTEM LGSTORE1 ZN GROUP
   **   ~~~   ** Method used to determine the cooling supply air flow rate is not specified when cooling coil is present.
   **   ~~~   ** Cooling Supply Air Flow Rate field is blank.
   **   ~~~   ** Cooling Supply Air Flow Rate Per Floor Area field is blank.
   **   ~~~   ** Cooling Fraction of Autosized Cooling Supply Air Flow Rate field is blank.
   **   ~~~   ** Cooling Supply Air Flow Rate Per Unit of Capacity field is blank.
   **   ~~~   ** Blank field not allowed for all four cooling supply air flow rate calculation methods when cooling coil is present.

   ** Warning ** AirLoopHVAC:UnitarySystem = FAN COOLING COIL HEATING COIL UNITARYSYSTEM LGSTORE1 ZN GROUP
   **   ~~~   ** Method used to determine the heating supply air flow rate is not specified when heating coil is present.
   **   ~~~   ** Heating Supply Air Flow Rate field is blank.
   **   ~~~   ** Heating Supply Air Flow Rate Per Floor Area field is blank.
   **   ~~~   ** Heating Fraction of Autosized Heating Supply Air Flow Rate field is blank.
   **   ~~~   ** Heating Supply Air Flow Rate Per Unit of Capacity field is blank.
   **   ~~~   ** Blank field not allowed for all four heating supply air flow rate calculation methods when heating coil is present.

**Comment by Nigusse** [comment]
@mjwitte @rraustad I added warning message. Should this type of input mistake be fatal, or warning is sufficient?

**Comment by mjwitte** [comment]
I would vote for warning.

**Comment by rraustad** [comment]
Well if all 4 inputs are blank for both cooling and heating and all rated flow rates = 0 then the system won't do anything. Would that ever be expected for an air system? Users sometimes don't look at the err file. I think if all air flow rates = 0 then that should be fatal (is there already a fatal for that condition?) ... or there should be a reason when an air system has 0 flow rate and then it could be just a warning.

**Comment by Nigusse** [comment]
The defect file has all unitary system (parent object) flow rates sizing fields and their flow sizing methods blank. Interestingly the file runs to completions without warning. On the other hand if you specify the flow sizing method and leave the corresponding flow sizing field blank, then it fatal out. I do not know a situation where all flow rates are zero.

**Comment by rraustad** [comment]
Leave it just a warning for now.

**Comment by nrel-bot-2c** [comment]
@Nigusse @lgentile it has been 28 days since this pull request was last updated.

**Comment by Nigusse** [comment]
This branch is ready for review.

**Comment by Nigusse** [comment]
Merged in develop and reolsved conflict.

File: src/EnergyPlus/UnitarySystem.cc [hunk 1]
```diff
@@ -6079,6 +6079,46 @@ namespace UnitarySystems {
...
+                                "Method used to determine the cooling supply air flow rate is not specified when cooling coil is present.");
+                            // check if all cooling flow calc method fields are blank
+                            if (((loc_m_CoolingSAFMethod_SAFlow == -999.0) && (loc_m_CoolingSAFMethod_SAFlowPerFloorArea == -999.0) &&
+                                 (loc_m_CoolingSAFMethod_FracOfAutosizedCoolingSAFlow == -999.0) &&
+                                 (loc_m_CoolingSAFMethod_FlowPerCoolingCapacity == -999.0))) {
+                                ShowContinueError("Cooling Supply Air Flow Rate field is blank.");
+                                ShowContinueError("Cooling Supply Air Flow Rate Per Floor Area field is blank.");
+                                ShowContinueError("Cooling Fraction of Autosized Cooling Supply Air Flow Rate field is blank.");
+                                ShowContinueError("Cooling Supply Air Flow Rate Per Unit of Capacity field is blank.");
+                                ShowContinueError("Blank field not allowed for all four cooling supply air flow rate calculation methods when "
```
Comment (mjwitte) [inline src/EnergyPlus/UnitarySystem.cc:6104]: It's a rather verbose warning, but that should get the user's attention.

**Review by mjwitte — APPROVED** [review]
Tested with defect file. The expected warning appears.
Will merge in develop and let CI run one more pass.

**Pull request merged**